cdc/processor/pipeline/sink.go

```diff
@@ -269,12 +268,11 @@ func (n *sinkNode) Receive(ctx pipeline.NodeContext) error {
 		}
 	case pipeline.MessageTypeCommand:
 		if msg.Command.Tp == pipeline.CommandTypeStopAtTs {
-			if msg.Command.StoppedTs < n.checkpointTs {
-				log.Warn("the stopped ts is less than the checkpoint ts, "+
-					"the table pipeline can't be stopped accurately, will be stopped soon",
-					zap.Uint64("stoppedTs", msg.Command.StoppedTs), zap.Uint64("checkpointTs", n.checkpointTs))
-			}
 			n.targetTs = msg.Command.StoppedTs
+			n.barrierTs = msg.Command.StoppedTs
```

Comment:
Why do we need to set `n.barrierTs`?

Reply:
if not  
```
	if resolvedTs > n.barrierTs {
		resolvedTs = n.barrierTs
	}
```

https://github.com/pingcap/ticdc/blob/78987aa2b736994b86a31d1c0c5848226720f2b2/cdc/processor/pipeline/sink.go#L119

Reply:
Is it possible that `resolvedTs > n.barrierTs`, if we don't manually set `n.barrierTs` explicit when stopping the table?

Reply:
according to the code, yes, this is a chance
https://github.com/pingcap/ticdc/blob/2fc6415c04560ed1e6bfb159ff4960bf01b22c65/cdc/processor/processor.go#L646
```
// pushResolvedTs2Table sends global resolved ts to all the table pipelines.
func (p *processor) pushResolvedTs2Table() error {
	resolvedTs := p.changefeed.Status.ResolvedTs
	for _, table := range p.tables {
		table.UpdateBarrierTs(resolvedTs)
	}
	return nil
}
```
https://github.com/pingcap/ticdc/blob/2fc6415c04560ed1e6bfb159ff4960bf01b22c65/cdc/processor/pipeline/table.go#L95